code/modules/mob/living/silicon/robot/drone/drone_items.dm

```diff
@@ -174,14 +303,37 @@
 
 	for(var/mob/M in T)
 		if(istype(M,/mob/living/simple_animal/lizard) || istype(M,/mob/living/simple_animal/mouse))
-			src.loc.visible_message("\red [src.loc] sucks [M] into its decompiler. There's a horrible crunching noise.","\red It's a bit of a struggle, but you manage to suck [M] into your decompiler. It makes a series of visceral crunching noises.")
+
+			src.loc.visible_message("\red [src.loc] begins to process the creature with its decompiler...","\red You begin to process the creature with your decompiler...")
+
+			if(!do_after(user,10))
+				user << "\red You need to remain still while decompiling such a large object."
+				return
+
+			src.loc.visible_message("\red [src.loc] sucks [M] into its decompiler. There's a horrible crunching and squelching noise.","\red It's a bit of a struggle, but you manage to suck [M] into your decompiler. It makes a series of visceral crunching and squelching noises.")
+			playsound(src.loc, 'sound/effects/gib.ogg', 30, 1, -2)
+			del(M)
 			new/obj/effect/decal/cleanable/blood/splatter(get_turf(src))
+			store_buffer["biomass"] += 25
+			grabbed_something = 1
+			continue
+
+		else if(istype(M,/mob/living/carbon) && M.stat == DEAD)
```

Comment:
Uh, What? Letting anything with a decompiler **DEL** any carbon mob? This isn't exempt from drones, and I can't even list the amount of things wrong with letting them **DEL** corpses, after only 5 seconds.